test/spmd/tensor/test_matrix_ops.py

```diff
@@ -127,6 +131,130 @@ def test_t(self):
+        batch_2 = torch.rand(3, 8, 6, device=self.device_type)
+
+        def test_placement_comb(
+            tensor_placements: Sequence[Placement],
+            batch_1_placements: Sequence[Placement],
+            batch_2_placements: Sequence[Placement],
+            beta: int,
+            alpha: int,
+        ) -> None:
+            tensor_dt = DTensor.from_local(
+                tensor, device_mesh, tensor_placements
+            )
+            batch_1_dt = DTensor.from_local(
+                batch_1, device_mesh, batch_1_placements
+            )
+            batch_2_dt = DTensor.from_local(
+                batch_2, device_mesh, batch_2_placements
+            )
+            new_dt = torch.baddbmm(
+                tensor_dt, batch_1_dt, batch_2_dt, beta=0.0, alpha=0.5
+            )
+            self.assertEqual(new_dt.to_local(), local_result)
+
+        shard0_spec = Shard(0)
+        shard1_spec = Shard(1)
+        shard2_spec = Shard(2)
+        shard_specs = [shard0_spec, shard1_spec, shard2_spec]
+        shard_specs_comb = list(
+            itertools.product(shard_specs, shard_specs, shard_specs)
+        )
+        passlist = []
+        numeric_params_comb = [
+            (0.0, 0.5),  # zero-beta
+            (0.8, 0.5),  # non-zero-beta
+        ]
+
+        def specInSeq(
+            s: Sequence[Placement], l: Sequence[Sequence[Placement]]
+        ) -> bool:
+            boolSeq = map(lambda r: list(s) == list(r), l)
+            return functools.reduce(lambda x, y: x or y, boolSeq, False)
+
+        for beta, alpha in numeric_params_comb:
+            local_result = torch.baddbmm(
+                tensor, batch_1, batch_2, beta=beta, alpha=alpha
+            )
+            # tests that currently pass
+            for spec in passlist:
+                test_placement_comb(
+                    [spec[0]], [spec[1]], [spec[2]], beta, alpha
+                )
+
+            # TODO: support these tests
+            shard_specs_comb = [
+                spec
+                for spec in shard_specs_comb
+                if not specInSeq(spec, passlist)
+            ]
+            for spec in shard_specs_comb:
+                with self.assertRaises(Exception):
+                    test_placement_comb(
+                        [spec[0]], [spec[1]], [spec[2]], beta, alpha
+                    )
+
+        # TODO: test with replicate
+
+    @with_comms
+    def test_sharded_bmm(self):
+        device_mesh = DeviceMesh(self.device_type, list(range(self.world_size)))
+        torch.manual_seed(self.rank)
+        input = torch.rand(3, 5, 8, device=self.device_type)
+        mat_2 = torch.rand(3, 8, 6, device=self.device_type)
+        local_result = torch.bmm(input, mat_2)
+
+        def test_placement_comb(
+            placements1: Sequence[Placement],
+            placements2: Sequence[Placement],
+        ) -> None:
+            input_dt = DTensor.from_local(input, device_mesh, placements1)
```

Comment:
I think your value inconsistency problem coming from the fact that you are using `from_local` and for the case [2, 1] you bmm it across local tensor boundaries, so i think you shouldn't use `from_local` here, can you just follow the `test_mm` and use `distribute_tensor` instead?
https://github.com/pytorch/tau/blob/main/test/spmd/tensor/test_matrix_ops.py#L86